tools/extract-locale.js

```diff
@@ -0,0 +1,73 @@
+var path = require('path');
+var fs = require('fs');
+var glob = require('glob');
+var merge = require('merge');
+
+function deconstructPath(filePath, chain) {
+    chain = chain || [];
+    if (chain[0] === 'bundles') {
+        return chain.slice(1, -3);
+    }
+    chain.unshift(path.basename(filePath));
+    return deconstructPath(path.dirname(filePath), chain);
+}
+
+function ensureDirectoryExists(filePath) {
+    var dirname = path.dirname(filePath);
+    if (fs.existsSync(dirname)) {
+      return true;
+    }
+    ensureDirectoryExists(dirname);
+    fs.mkdirSync(dirname);
```

Comment:
Shouldn't mkdir and recursion be switched?

Reply:
We are recursing from the end of the path down (right to left). By calling the self-recursion first, we first make sure all parent directories exist. After that's done, we can create the `dirname` on the "current" level.

If we were recursing from start to end (left to right), then we would first do `mkdirSync`.